homeassistant/components/google_photos/coordinator.py

```diff
@@ -0,0 +1,70 @@
+"""Coordinator for fetching data from Google Photos API.
+
+This coordinator fetches the list of Google Photos albums that were created by
+Home Assistant, which for large libraries may take some time. The list of album
+ids and titles is cached and this provides a method to refresh urls since they
+are short lived.
+"""
+
+import asyncio
+import datetime
+import logging
+from typing import Final
+
+from google_photos_library_api.api import GooglePhotosLibraryApi
+from google_photos_library_api.exceptions import GooglePhotosApiError
+from google_photos_library_api.model import Album
+
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+UPDATE_INTERVAL: Final = datetime.timedelta(hours=24)
+ALBUM_PAGE_SIZE = 50
+
+
+class GooglePhotosUpdateCoordinator(DataUpdateCoordinator[dict[str, str] | None]):
```

Comment:
why is it nullable?

Reply:
The idea was to handle the not yet initialized state here. I realize this should be a thing that all coordinators need to worry about but I didn't see it handled in the parent class typing anywhere. Maybe I missed it.

Reply:
why not overwrite the `self.data` in the constructor to `{}`?

Reply:
Having no albums is also a valid state and I want to know the difference. This may only matter to:
(1) Know if we should render an empty album page or fail since it's not yet loaded
(2) in follow up use cases described in the PR to know if an album can't be found vs not loaded.